src/CoreWCF.RabbitMQ.Client/src/CoreWCF/ServiceModel/Channels/RabbitMqChannelFactory.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Linq;
+using System;
+using System.ServiceModel.Channels;
+using System.Threading.Tasks;
+
+namespace CoreWCF.ServiceModel.Channels
+{
+    public class RabbitMqChannelFactory : ChannelFactoryBase<IOutputChannel>
+    {
+        private BufferManager _bufferManager;
+        private MessageEncoderFactory _messageEncoderFactory;
+        private RabbitMqTransportBindingElement _transport;
+
+        internal RabbitMqChannelFactory(RabbitMqTransportBindingElement transport, BindingContext context)
+            : base(context.Binding)
```

Comment:
You should get the client credentials in the constructor of this ChannelFactory. You can see the code to do this [here](https://source.dot.net/#System.ServiceModel.Http/System/ServiceModel/Channels/HttpChannelFactory.cs,129) from HttpChannelFactory. You then get the SecurityTokenManager like [here](https://source.dot.net/#System.ServiceModel.Http/System/ServiceModel/Channels/HttpChannelFactory.cs,557) and then get the token provider for the credentials. I suggest using ClientCredentials.UserName as the property to get the credentials from. If you do that, you can see the call to a helper method that Http uses for the UserName credentials [here](https://source.dot.net/#System.ServiceModel.Http/System/ServiceModel/Channels/HttpChannelFactory.cs,461). The implementation is [here](https://source.dot.net/#System.ServiceModel.Primitives/System/ServiceModel/Channels/TransportSecurityHelpers.cs,170). You generally make the call to SecurityTokenManager.CreateSecurityTokenProvider during the call to opening the channel as there a ChannelParameterCollection that can be passed like this:
```c#
var channel = factory.CreateChannel();
var channelParameters = ((IChannel)channel).GetProperty<ChannelParameterCollection>();
channelParameters.Add(new Foo());
((IChannel)channel).Open(); // Causes the security token provider to be created using the channel parameters
```
This means the actual IOutputChannel is the one that needs to get the token provider, and the security token on the open call path..